xonsh/tools.py

```diff
@@ -92,6 +93,85 @@ def __init__(self, returncode, command, output=None, stderr=None,
         self.completed_command = completed_command
 
 
+class EnvPath(MutableSequence):
+    """
+    A class that implements an environment path, which is a list of
+    strings. Provides a custom method that expands all paths if the
+    relevant env variable has been set.
+    """
+    @staticmethod
+    def _expandpath(path):
+        """
+        Performs environment variable / user expansion on a given path
+        if the relevant flag has been set.
+        """
+        env = getattr(builtins, '__xonsh_env__', os.environ)
+        if env.get('EXPAND_ENV_VARS', False):
+            # expand variables and use os.path.realpath to handle cases
+            # with relative paths like ../ or ./
+            path = os.path.realpath(os.path.expanduser(expandvars(path)))
+        return path
+
+    @staticmethod
+    def _decode_bytes(path):
```

Comment:
These two methods probably should be at global scope, rather than being staticmethods. They don't deal with the class at all.